### ISSUE TITLE
Steam games all deleted

### ISSUE THREAD

**Issue by saward** [ValveSoftware/steam-for-linux#3409]
So it appears that steam has deleted all my games.  I've done a space visualisation of my home folder, and I cannot see them anywhere.  The space used by the files found seems to match that which df -h reports as used, so I don't think they're hiding anywhere else.

Was using gnome 3 shell, and had some issues.  Couldn't move a window without the shell appearing to no longer be interactive.  Restarted X a couple of times with ctrl-alt-del.  May or may not have had something to do with me trying to use:
`LD_PRELOAD="libpthread.so.0 libGL.so.1" __GL_THREADED_OPTIMIZATIONS=1`

Anyway, finally started up steam and it appeared to be updating and failing.  Cancelled since it took a while a couple of times, and started.  Ran it from the console instead so I could see what happened.  Waiting patiently lead to the update completing, and the discovery that my games were missing.  Console output available at https://gist.github.com/saward/ecd01a8f56656d0b0110

**Comment by nortexoid** [comment]
Same thing just happened to me. The updater was stuck, so I canceled it. I then reopened Steam and was greeted by the login screen, even though it's set to automatically login. All my installed games were missing (the whole folder ~/.steam/steam/steamapps/common/ was missing).

Fortunately I back everything up semi-regularly so it wasn't much of a hassle, but damn!

**Comment by enricoluigi** [comment]
Just happened to me, steamapps folder has been deleted 
now i cant run steam, "unable to connect to x"

**Comment by LBEaston** [comment]
Chiming in to say it happened to me as well. Hit cancel on the updater, logged in, games are gone.

**Comment by BenjaminSchaaf** [comment]
Just wanted to note: This happened to me as well. Pressed cancel on the updater. Once it opened again it seemed that two instances of the updater opened. One hung while the other completed. After a error message about missing 32bit libraries. Then steam started as normal but did not remember my login details and all my games are now gone.

**Comment by nortexoid** [comment]
Come on, this just happened to me again! There wasn't even anything unusual like an updater issue. It wouldn't be such a problem if it could rediscover my backup and not have to download everything from the internet again.

**Comment by JamieOnUbuntu** [comment]
I've noticed that if you ever open Steam and it doesn't automatically log you in, logging in will cause the deletion of your games. If you aren't automatically logged in, close the login box and fully exit Steam. When you launch Steam again it should log you in and your games will still be there.

Cancelling the auto updater seems to always delete your games.

**Comment by JohnMcLear** [comment]
Steam updated and deleted all my games so +1.

**Comment by Martinsos** [comment]
Same happens to me on archlinux, after each update all games get deleted!

**Comment by LegitGamez** [comment]
Good to know i'm not like Rhianna being the only one with this in the world. For me it crashed when I tried to launch ArmA 3 and I couldn't verify to get a missing exe. Steam crashed and didn't automatically log me in. BOOM! Gone! 10 times!

**Comment by ashishmadeti** [comment]
Happened to me 3 times now. Its very annoying.
I am on Kubuntu 15.04 and my steam folder is symlinked to a folder in my external drive.

**Comment by kmf** [comment]
Steam updated and deleted all my games so +1.

**Comment by WhiteWolf753** [comment]
Steam just delete GTA V and Tomb Raider. It's 3 times now. GG!

**Comment by Tele42** [comment]
@WhiteWolf753 what OS are you on? This issue report was limited to linux reports, but if there's a cross platform issue, that might lead a valve dev to check elsewhere in the client.

**Comment by Translator5** [comment]
@WhiteWolf753 if you use wine, it could be because of your unplugged hard-drive, where your games are installed!
Try adding a new game-library. It fixed it for me!

**Comment by WhiteWolf753** [comment]
I have Windows 10 Pro x64

**Comment by Translator5** [comment]
@WhiteWolf753 This here is about Steam for Linux :laughing:

**Comment by FeralBytes** [comment]
I too was affected by this... restored the games by mv the steamapps folder from .save but in doing so lost my saved games....

**Comment by Martinsos** [comment]
This is not a problem for me any more - Steam updates now without deleting any games.

**Comment by FeralBytes** [comment]
I too was affected by this... restored the games by mv the steamapps folder from .save but in doing so lost my saved games....

**Comment by robertc** [comment]
Just happened to me: just finished downloading Mad Max, restarted Steam, all games gone from disk.  Additionally Steam is now adamant it wants to put my games into ~/Steam instead of ~/.local/share/Steam.  It seems to be failing to pick up directories in the launch script, I kept getting an error at line 94:

```
Setting up Steam content in /.local/share/Steam
mkdir: cannot create directory ‘/.local’: Permission denied
/usr/bin/steam: line 98: cd: /.local/share/Steam: No such file or directory
```

I had to hardcode the script to use ~/.local/share/Steam to get it to run at all.

**Comment by thefarwind** [comment]
hmm, just chiming in to say this has happened to me as well.

**Comment by redog** [comment]
this happened to me. Updated my nvidia-drivers and tried to launch steam....steam ran in the background but never would launch...after I rebooted steam launched but deleted the folder with all the game content and any custom class configurations and binds I had.....this is the third time something has happened causing me to download the entire game library again.

**Comment by mrbobaking** [comment]
this just happened to me for the third time. When it happened for the first and the second time, I was playing CS:GO and the game just froze. I pressed ctrl+alt+f1 and ubuntu gave an error and deleted the Steam folder. third time, however, steam wouldn't open so I killed the process and folder got emptied but with a .save folder inside. And the .save folder had the exact copy of the Steam folder. just like @Pr0b described.

**Comment by cfabyss** [comment]
This happened to me today, too. I checked the .save directory, but nothing was there. I did attempt to verify integrity of game cache for the only games I had installed - maybe this has something to do with why .save was not populated?

**Comment by RaJiska** [comment]
Still happening here, and for quite a while. No .save file could be found.

**Comment by Matz05** [comment]
This just happened to me today as well. I think I may have cancelled an auto-update with the 'cancel' button at some point (which I would expect would leave things in a valid state and retry the installation later), and then Steam spends an awful long time updating next time, requires me to re-authenticate my computer as 'a new computer' (which is odd in itself), and then I find out that my entire library has been deleted, and that this is a known, very old problem with their Linux client which they refused to fix for all this time. Trying the manual restore from the .save folder now, but first I'm backing it up to an external disk because I lack the disk space for a second copy. Very disappointed, this final straw has codified the plan I was already following: I don't plan to buy anything from Valve Corporation ever again. DRM-free only for me.

Running Linux here, of course, as the other sufferers mostly seem to.

**Comment by miguev** [comment]
This just happened to me, still a thing in 2018 on Ubuntu Studio 17.10.

TL;DR: rm -f .steampid .steam/steam.pid .steam/steam.pipe

My 100+ games were still installed taking 360G+ under .local/share/Steam/ ─ as you can imagine I was not looking forward to reinstalling even one of them, let alone losing all my progress.

Instead, in the old traditions of Un*x things that crash and then break, Steam seems to have left behind residual files that somehow tricked it into misbehaving. Other applications abort with a clear error message telling you to delete such files, Steam does not. So I just exit Steam, delete those files above, then start Steam again and all my installed games (and progress in them) are back!

Note: I did not interrupt the updater or forced Steam to end with "Exit now" or anything like that, but I am having more kernel lockups ever since the patches for Meltdown, may be just coincidence but the timing aligns.

**Comment by kmf** [comment]
Just happened to me. Again.

**Comment by rlka** [comment]
Yup, have the same problem now. Just started my steam as always, message "Updating User Configuration" was shown and no settings could be saved now.

**Comment by shimajan** [comment]
Launching Steam on Linux just deleted the entirety of Rainbow Six Siege from my hard drive.
It takes many hours preallocating to reinstall, very sad.

**Comment by Tviljan** [comment]
Also files deleted from several games. Here sis content_log.txt

`[2020-12-06 17:12:32] Loaded 10 apps from install folder "/home/xx/.local/share/Steam/steamapps\appmanifest_*.acf".
[2020-12-06 17:12:32] Loaded 2 apps from install folder "/home/xx/.local/share/SteamLibrary/steamapps\appmanifest_*.acf".
[2020-12-06 17:12:35] AppID 32370 scheduler update : Priority Last, timeSinceLastPlayed=15460, appDisableSecondsRemaining=0
[2020-12-06 17:12:35] AppID 32370 state changed : Update Required,Fully Installed,Update Queued,
[2020-12-06 17:12:35] AppID 358270 scheduler update : Priority Last, timeSinceLastPlayed=14024, appDisableSecondsRemaining=0
[2020-12-06 17:12:35] AppID 358270 state changed : Update Required,Fully Installed,Update Queued,
[2020-12-06 17:12:35] AppID 32370 state changed : Update Required,Fully Installed,Update Queued,Update Running,
[2020-12-06 17:12:35] AppID 32370 update changed : Running,
[2020-12-06 17:12:35] AppID 32370 update changed : Running,Reconfiguring,
[2020-12-06 17:12:35] AppID 32370 update changed : Running,
[2020-12-06 17:12:35] AppID 32370 update changed : Running,Validating,
[2020-12-06 17:12:35] AppID 32370 update changed : Running,
[2020-12-06 17:12:35] AppID 32370 update started : download 0/0, store 0/0, reuse 0/0, delta 0/0, stage 0/0 
[2020-12-06 17:12:35] AppID 32370 has no changes, active: 32371 (4122010938027300909), target: 
[2020-12-06 17:12:35] AppID 32370 state changed : Update Required,Fully Installed,Update Queued,Update Running,Update Started,
[2020-12-06 17:12:35] AppID 32370 update changed : Running,Committing,
[2020-12-06 17:12:35] AppID 32370 starting commit from "/home/xx/.local/share/SteamLibrary/steamapps/downloading/32370" to "/home/xx/.local/share/SteamLibrary/steamapps/common/swkotor" : 0 updated, 0 moved, 16044 deleted files
[2020-12-06 17:12:37] AppID 32370 state changed : Fully Installed,Update Queued,Update Running,
[2020-12-06 17:12:37] AppID 32370 finished update, mounted depots (BuildID 4143502) : 
[2020-12-06 17:12:37] AppID 32370 update changed : Running,
[2020-12-06 17:12:37] AppID 32370 update changed : None
[2020-12-06 17:12:37] AppID 32370 state changed : Fully Installed,Update Queued,
[2020-12-06 17:12:37] AppID 32370 state changed : Fully Installed,
[2020-12-06 17:12:37] AppID 32370 scheduler finished : removed from schedule
[2020-12-06 17:12:38] AppID 358270 state changed : Update Required,Fully Installed,Update Queued,Update Running,
[2020-12-06 17:12:38] AppID 358280 state changed : Fully Installed,Shared Only,Update Running,`

**Comment by burghoffdavid** [comment]
How is this 8 (!!!) year old issue still not fixed. Was about to make the full switch to linux for gaming, and now this happens. Completely unacceptable, back to windows and GOG I guess, wp valve :clap:
